anyio/__init__.py

```diff
@@ -532,6 +533,11 @@ async def create_udp_socket(
     raw_socket = socket.socket(family=family, type=socket.SOCK_DGRAM)
     sock = _get_asynclib().Socket(raw_socket)
     try:
+        if sys.platform == 'win32':
```

Comment:
These can already be set after creation. Why should we change the default behavior?

Reply:
Because they need to be set before calling `bind()`. Some OSes don't accept calling that twice.

Reply:
But I thought you said in the beginning that you can't `bind()` to an actual address? What am I missing here?

Reply:
You can't bind to an (IPv4/IPv6) address, but you have to bind to the actual (UDP) port the multicast packets are addressed to. Otherwise you don't get them.